sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1069,11 +1110,19 @@ class Analyzer(
         case a @ Aggregate(grouping, expressions, child) =>
           failOnOuterReference(a)
           val referencesToAdd = missingReferences(a)
+          if (foundNonEqualCorrelatedPred) {
+            failOnNonEqualCorrelatedPredicate(a)
+          }
           if (referencesToAdd.nonEmpty) {
             Aggregate(grouping ++ referencesToAdd, expressions ++ referencesToAdd, child)
           } else {
             a
           }
+        case w : Window =>
+          if (foundNonEqualCorrelatedPred) {
+            failOnNonEqualCorrelatedPredicate(w)
+          }
+          w
```

Comment:
Should we rewrite the aggregates in the window function? For example:
```sql
select *
from  tbl_x
where value in (select max() over() from tbl_y where tbl_y.id = tbl_x.id)
```
Should we rewrite `max() over()` into `max() over(partition by tbl_y.id)`?